### DISCUSSION TITLE
React UI Plugin development

### DISCUSSION THREAD

**Discussion by tdesveaux** [buildbot/buildbot#7086] [Q&A] [answered]
Hi!

I'm working with a pretty massive configuration and would like to customize the UI so it's tailored to our usage.

I managed to pretty easily duplicate the react-grid-view plugin and hack around it.
My workflow was ... not great. I'm not familiar with react and web frontend dev in general so I ended up running a `yarn run build` and moving the build artifacts into the installed plugin directory.
It was acceptable when trying things out, but I'd like a better workflow moving forward where I'm able to break/debug and all.

So here's my question. Is there or will there be documentation on react plugin dev for Buildbot?

Something else I'm interested in, is it possible for the Buildbot configuration to provide configuration to frontend plugins?
`master/buildbot/www/plugin.py:Application` which UI plugins are based on seems to be doing nothing with the provided configuration AFAIK?
`registerSettingGroup` looks to be more about user settings?

I expect @p12tic to have these information (sorry for the ping!)

**Comment by p12tic** [comment, marked as answer]
React plugin development workflow is not yet fleshed out because the React frontend is still under development. Once it's the primary frontend there will be more focus on making plugin development for React frontend more straightforward.

**Reply by tdesveaux** [reply]
Thanks for the answer.

I made a repository for this plugin [here](https://github.com/tdesveaux/DNE-Buildbot-React-grid-view-plugin) if you are curious about it (well, it needs some refining).

I tweaked a bit the [build-dev](https://github.com/tdesveaux/DNE-Buildbot-React-grid-view-plugin/blob/main/package.json#L10) command to include sourcemap and remove minification which helped for things like callstack in the browser.
But still had to fallback de console.log for debug some times.

A last question if you don't mind. What would you say the state of the React frontend is ? I found some minor issues but it seems overall pretty usable.
